lib/streamlit/elements/image.py

```diff
@@ -218,374 +175,3 @@ def image(
-
-        # Following regex allows svg image files to start either via a "<?xml...>" tag
-        # eventually followed by a "<svg...>" tag or directly starting with a "<svg>" tag
-        if re.search(r"(^\s?(<\?xml[\s\S]*<svg\s)|^\s?<svg\s|^\s?<svg>\s)", image):
-            if "xmlns" not in image:
-                # The xmlns attribute is required for SVGs to render in an img tag.
-                # If it's not present, we add to the first SVG tag:
-                image = image.replace(
-                    "<svg", '<svg xmlns="http://www.w3.org/2000/svg" ', 1
-                )
-            # Convert to base64 to prevent issues with encoding:
-            import base64
-
-            image_b64_encoded = base64.b64encode(image.encode("utf-8")).decode("utf-8")
-            # Return SVG as data URI:
-            return f"data:image/svg+xml;base64,{image_b64_encoded}"
-
-        # Otherwise, try to open it as a file.
-        try:
-            with open(image, "rb") as f:
-                image_data = f.read()
-        except Exception:
-            # When we aren't able to open the image file, we still pass the path to
-            # the MediaFileManager - its storage backend may have access to files
-            # that Streamlit does not.
-            import mimetypes
-
-            mimetype, _ = mimetypes.guess_type(image)
-            if mimetype is None:
-                mimetype = "application/octet-stream"
-
-            url = runtime.get_instance().media_file_mgr.add(image, mimetype, image_id)
-            caching.save_media_data(image, mimetype, image_id)
-            return url
-
-    # PIL Images
-    elif isinstance(image, (ImageFile.ImageFile, Image.Image)):
-        format = _validate_image_format_string(image, output_format)
-        image_data = _PIL_to_bytes(image, format)
-
-    # BytesIO
-    # Note: This doesn't support SVG. We could convert to png (cairosvg.svg2png)
-    # or just decode BytesIO to string and handle that way.
-    elif isinstance(image, io.BytesIO):
-        image_data = _BytesIO_to_bytes(image)
-
-    # Numpy Arrays (ie opencv)
-    elif isinstance(image, np.ndarray):
-        image = _clip_image(
-            _verify_np_shape(image),
-            clamp,
-        )
-
-        if channels == "BGR":
-            if len(cast(NumpyShape, image.shape)) == 3:
-                image = image[:, :, [2, 1, 0]]
-            else:
-                raise StreamlitAPIException(
-                    'When using `channels="BGR"`, the input image should '
-                    "have exactly 3 color channels"
-                )
-
-        # Depending on the version of numpy that the user has installed, the
-        # typechecker may not be able to deduce that indexing into a
-        # `npt.NDArray[Any]` returns a `npt.NDArray[Any]`, so we need to
-        # ignore redundant casts below.
-        image_data = _np_array_to_bytes(
-            array=cast("npt.NDArray[Any]", image),  # type: ignore[redundant-cast]
```

Comment:
MyPy complained that the ignore statement here is unused, so I removed it (as well as the comment above) in `image_utils.py`. Not entirely sure though if this is still needed for some very old numpy versions?

Reply:
Yeah it looks like this was added 2 years ago in https://github.com/streamlit/streamlit/pull/4761, so I hope its safe to remove now.
Based on the comment it sounds like we might be able to remove the cast as well as it is described as redundant. Could you try this out and see whether anything complains?

Reply:
Removed the cast and everything seems to work fine, so I think it's probably fine to leave it like this. I can't imagine a realistic scenario where someone would use the latest Streamlit version but a NumPy version that's older than 2 years.